[Reviewer: Ellie] Add etcd scripts for N-site GR upgrade

sebrexmetaswitch opened the pull request:
The scripts needed for the N-Site GR upgrade.

We only need 3, because the existing load_from_*_cluster scripts should work fine for populating etcd with the cluster info after merging Sprout and Ralf Chronos/memcached clusters

Testing done: manual testing that the correct etcd config is copied/saved/loaded

eleanor-merry commented on clearwater-etcd/usr/share/clearwater/clearwater-etcd/scripts/save_etcd_config.py:
```diff
@@ -0,0 +1,121 @@
+# Produces a dictionary like {"[node_ip]": "normal"} containing every cassandra
+# node in the local site
+def get_local_cassandra_nodes(site_name):
+    nodetool_process = subprocess.Popen("nodetool status", stdout=subprocess.PIPE, shell=True)
```
This needs to use the signalling namespace (use /usr/share/clearwater/bin/run-in-signaling-namespace, but please check the path)

eleanor-merry replied:
This still needs doing

eleanor-merry commented on clearwater-etcd/usr/share/clearwater/clearwater-etcd/scripts/save_etcd_config.py:
```diff
@@ -0,0 +1,121 @@
+# Now save the dictionary of keys we want to preserve
+save_to_disk(data_to_save)
+
+print("\nSaved etcd cluster info to disk")
```
It would be nice to easily see what data has been saved, so we can check it. What do you think?

eleanor-merry commented on clearwater-etcd/usr/share/clearwater/clearwater-etcd/scripts/load_etcd_config.py:
```diff
@@ -0,0 +1,52 @@
+
+# Add to etcd
+for key, value in data.iteritems():
+    subprocess.check_output(["clearwater-etcdctl", "set", key, value])
```
It would be nice to see what data we're about to import before doing so. What do you think?

sebrexmetaswitch commented on clearwater-cluster-manager.root/usr/share/clearwater/clearwater-cluster-manager/scripts/recreate_cluster:
```diff
@@ -0,0 +1,52 @@
+
+set -ue
+
+local_site_name=site1
```
No, left that in by mistake.

sebrexmetaswitch commented:
As discussed, added some more checks for etcdctl errors, and allowed the user to specify the location that the config is saved to.
Also split the recreate_cluster script into homestead and sprout versions to reduce the chance of user error.

eleanor-merry commented on clearwater-cluster-manager.root/usr/share/clearwater/clearwater-cluster-manager/scripts/recreate_homestead_cluster:
```diff
@@ -0,0 +1,48 @@
+if ! nc -z ${management_local_ip:-$local_ip} 4000
+then
+  echo "The Clearwater Configuration store (etcd) is not running"
+  echo "Please start it before uploading configuration"
```
This isn't a relevant error message (uploading configuration?). Can you fix this throughout?

eleanor-merry commented on clearwater-etcd/usr/share/clearwater/clearwater-etcd/scripts/save_etcd_config.py:
```diff
@@ -0,0 +1,136 @@
+            # indicating that the following nodes are part of the local site
+            if "Datacenter: {0}".format(site_name) in line:
+                found_local_section = True
+
```
Extra blank line
